### PR TITLE
Multiplayer & ChannelAPI fundamentals

### PR DESCRIPTION
**Pull request by blanche**
implement fundamentals of multiplayer, channel API and deployment changes
you can join a game from the home screen another player has created and not started yet (e.g. not made a guess yet)
guesses from other players are then pushed to all clients to

please do not just approve this request but test it, there were a lot of merge errors

EVERYBODY MUST READ:
- for this to work, the game has to be deployed on the python server, as it will be in production
- in js-src, run `npm run dev-build`, this will create the sources to /python-src/webapp and keep it up to date as with the dev server
- run the python server as usual, calling http://localhost:8081/ will then be the url of the frontend AND the backend (with /api/)

missing:
- make join game an extra step in the workflow between create player and game (as discussed in meeting before the holidays)
- other guesses are currently only shown in console - display them on the map
- tons of error handling
- make it pretty
- synchronise players always playing in the same round (especially when displaying the highscore)

to play:
- create a game as usual, but don't make a guess yet!
- as player 2 (second browser), search for games on the home screen and join
- you should now play the same cities
- watch the console to see what other guesses the player made

i disabled the game countdown, because player 1 has to wait for player 2 to join and the countdown would make this kind of stressful

### CODE REVIEW

File: python-src/resources.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import datetime
 import random
 
+from google.appengine.api.channel import channel
```
Comment (andreaseliasson) [inline python-src/resources.py:4]: Do we need to install a package here, and if so which one?

Comment (andreaseliasson) [reply python-src/resources.py:4]: I tried pip installing the `google-api-python-client` and I now have this in my lib folder.

screen shot 2016-12-27 at 14 39 53

What am I missing here? (I can't import `google.appengine.api.channel`)

**Comment by blanche** [comment]
This should be included in the python gae sdk, I didn't install anything additionally

File: python-src/resources.py (hunk 1 above)
Comment (blanche) [reply python-src/resources.py:4]: Did you manage to get it work? We can make a quick session tomorrow morning if you are free

Comment (andreaseliasson) [reply python-src/resources.py:4]: Yes, I was able to get it working 👍

File: python-src/webapp [hunk 2]
```diff
@@ -0,0 +1 @@
+/home/lukas/uni/pintheworld/js-src/dist
```
Comment (andreaseliasson) [inline python-src/webapp:1]: Should we remove this directory from git? Running `npm run build-dev` throws an exception the first time. Removing this folder and running `npm run build-dev` fixes the problem.

**Review by andreaseliasson — CHANGES_REQUESTED** [review]
Should address what to do with the `python-src/webapp` dir.

**Review by andreaseliasson — COMMENTED** [review]
Playing multi-player works for me.

**Review by andreaseliasson — APPROVED** [review]

File: js-src/src/app/map.component.js [hunk 3]
```diff
@@ -17,167 +22,186 @@ let MapComponent = Component({
...
-					this.gameEnded = true;//Game ended, do not initialize the counter for rounds
-					alert("Total Score: " + self.currentScore);
-					self.router.navigate(['/highscores', self.game.id, self.player.id ])//Move to highscores when the game ends
-				}
-			}
+            if (!this.responseTaken) {//If map is already clicked, dont let it be clicked again
+                this.roundTimer = 0;//Set round timer to 0, round ended
+                self.currentScore += guess.score;
+                // TODO: push different colored markers
+                self.markers.push({lat: currentCity.lat, lng: currentCity.long});
```
Comment (JaneJiangHH) [inline js-src/src/app/map.component.js:152]: What is this line of code used for? markers is used to push marker where player clicked; cityMarkers is to push the right marker where the city is.

**Pull request merged**